services/app-api/utils/changeUserStatus.js

```diff
@@ -24,57 +24,79 @@ export const changeUserStatus = async ({
+      "#doneByEmail = :doneByEmail, #doneByName = :doneByName, " +
+      "#role = :role, #territory = :territory, #date = :date, " +
+      "#GSI1pk = :GSI1pk, #GSI1sk = :GSI1sk, #GSI2pk = :GSI2pk, #GSI2sk = :GSI2sk",
+    ExpressionAttributeNames: {
+      "#email": "email",
+      "#status": "status",
+      "#fullName": "fullName",
+      "#doneByEmail": "doneByEmail",
+      "#doneByName": "doneByName",
+      "#role": "role",
+      "#territory": "territory",
+      "#date": "date",
+      "#GSI1pk": "GSI1pk",
+      "#GSI1sk": "GSI1sk",
+      "#GSI2pk": "GSI2pk",
+      "#GSI2sk": "GSI2sk",
+    },
+    ExpressionAttributeValues: {
+      ":email": email.toLowerCase(),
+      ":status": status,
+      ":fullName": fullName,
+      ":doneByEmail": doneByEmail.toLowerCase(),
+      ":doneByName": doneByName,
+      ":role": role,
+      ":territory": territory,
+      ":date": date,
+      ":GSI1pk": "USER",
+      ":GSI1sk": buildSK(role, territory),
+      ":GSI2pk": `${role}#${territory}`,
+      ":GSI2sk": status,
+      ":incrval": 1,
+    },
+  };
+  const getParams = {
+    TableName: process.env.oneMacTableName,
+    Key: {
+      pk: email.toLowerCase(),
+      sk: `v0#${role}#${territory}`,
+    },
+  };
+
   try {
-    const updateParams = {
-      TableName: process.env.oneMacTableName,
-      ReturnValues: "UPDATED_NEW",
-      Key: {
-        pk: email.toLowerCase(),
-        sk: `v0#${role}#${territory}`,
-      },
-      UpdateExpression:
-        "SET Latest = if_not_exists(Latest, :defaultval) + :incrval, " +
-        "#email = :email, #status = :status, #fullName = :fullName, " +
-        "#doneByEmail = :doneByEmail, #doneByName = :doneByName, " +
-        "#role = :role, #territory = :territory, #date = :date, " +
-        "#GSI1pk = :GSI1pk, #GSI1sk = :GSI1sk, #GSI2pk = :GSI2pk, #GSI2sk = :GSI2sk",
-      ExpressionAttributeNames: {
-        "#email": "email",
-        "#status": "status",
-        "#fullName": "fullName",
-        "#doneByEmail": "doneByEmail",
-        "#doneByName": "doneByName",
-        "#role": "role",
-        "#territory": "territory",
-        "#date": "date",
-        "#GSI1pk": "GSI1pk",
-        "#GSI1sk": "GSI1sk",
-        "#GSI2pk": "GSI2pk",
-        "#GSI2sk": "GSI2sk",
-      },
-      ExpressionAttributeValues: {
-        ":email": email.toLowerCase(),
-        ":status": status,
-        ":fullName": fullName,
-        ":doneByEmail": doneByEmail.toLowerCase(),
-        ":doneByName": doneByName,
-        ":role": role,
-        ":territory": territory,
-        ":date": date,
-        ":GSI1pk": "USER",
-        ":GSI1sk": buildSK(role, territory),
-        ":GSI2pk": `${role}#${territory}`,
-        ":GSI2sk": status,
-        ":defaultval": 0,
-        ":incrval": 1,
-      },
-    };
+    // fix for latest being null or undefined
+    const result = await dynamoDb.get(getParams);
+    if (typeof result.Item?.Latest !== "number") {
```

Comment:
Again im not following the changes here because the original implementation used if_not_exists which would cover null and undefined use cases.

Reply:
ok relooking i do see that this use case would also cover invalid character data that is not numerical and not just null or undefined